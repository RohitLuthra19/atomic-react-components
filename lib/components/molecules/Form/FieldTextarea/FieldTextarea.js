// @flow
import React from 'react';
import type { Node } from 'react';
import { Field, ErrorMessage } from 'formik';
import styled from 'styled-components';
import Textarea from '../../../atoms/Textarea';
import FieldError from '../../../atoms/FieldError';
import Label from '../../../atoms/Label';
import styles from './FieldTextarea.style';

type Props = {
  id: string,
  className?: string,
  name: string,
  label: Node,
  labelProps?: {
    className?: string,
  },
  value?: string,
};

const FieldTextarea = ({
  id,
  className,
  type,
  name,
  label,
  labelProps,
  ...others
}: Props): Node => (
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>
      {label}
    </Label>
    <Field id={id} name={name} render={({ field }) => <Textarea {...others} {...field} />} />
    <ErrorMessage name={name} component={FieldError} />
  </div>
);

FieldTextarea.defaultProps = {
  className: '',
  labelProps: {},
  value: '',
};

export default styled(FieldTextarea)`
  ${styles};
`;
export { FieldTextarea as FieldTextareaVanilla };
