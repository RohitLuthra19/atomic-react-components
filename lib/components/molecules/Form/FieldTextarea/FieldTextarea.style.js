import { css } from 'styled-components';

const styles = css`
  margin-bottom: ${props => props.theme.spacing.unit1};

  label,
  textarea {
    display: block;
    margin-bottom: ${props => props.theme.spacing.halfUnit};
  }
`;

export default styles;
