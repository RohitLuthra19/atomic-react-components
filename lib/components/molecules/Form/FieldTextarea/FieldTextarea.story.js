// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import FieldTextarea from '.';
import Form from '../Form';
import defaultTextareaForm from './FieldTextarea.mock';

storiesOf('Molecules', module).addWithChapters('FieldTextarea', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <Form {...defaultTextareaForm}>
              <FieldTextarea
                id="comment"
                name="comment"
                label="Comment"
                placeholder="Enter your text here..."
              />
            </Form>
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
  ],
});
