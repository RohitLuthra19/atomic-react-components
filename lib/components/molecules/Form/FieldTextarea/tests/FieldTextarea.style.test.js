import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import FieldTextarea from '../FieldTextarea';

describe('FieldTextarea', () => {
  test('should render correctly', () => {
    expect(shallow(<FieldTextarea />)).toMatchSnapshot();
  });
});
