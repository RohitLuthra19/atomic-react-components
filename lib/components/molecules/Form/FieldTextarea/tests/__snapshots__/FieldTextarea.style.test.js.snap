// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldTextarea should render correctly 1`] = `
.c0 {
  margin-bottom: 1rem;
}

.c0 label {
  display: block;
  margin-bottom: 0.25rem;
}

<FieldTextarea
  className="c0"
  labelProps={Object {}}
  value=""
/>
`;
